scio-repl/src/main/scala/com/spotify/scio/repl/ScioILoop.scala

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2016 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.repl
+
+import java.io.BufferedReader
+
+import scala.tools.nsc.GenericRunnerSettings
+import scala.tools.nsc.interpreter.{IR, JPrintWriter}
+
+/**
+ * ScioILoop - core of Scio REPL
+ * @param scioClassLoader [[ScioReplClassLoader]] used for runtime classloading
+ * @param args user arguments for Scio REPL
+ * @param in [[ILoopCompat]] input
+ * @param out [[ILoopCompat]] output
+ */
+class ScioILoop(scioClassLoader: Option[ScioReplClassLoader],
+                args: List[String],
+                in: Option[BufferedReader],
+                out: JPrintWriter)
+  extends ILoopCompat(in, out) {
+
+  def this(scioCL: ScioReplClassLoader, args: List[String]) = this(Some(scioCL), args, None, new JPrintWriter(Console.out, true))
+
+  settings = new GenericRunnerSettings({ s => echo(s) })
+
+  override def printWelcome() {
+    echo("Welcome to Scio REPL!")
+  }
+
+  private val createJarCmd = LoopCommand.nullary("createJar", "creates Scio REPL runtime jar",
+    () => { Result.resultFromString(scioClassLoader.get.createReplCodeJar) } )
+  private val getNextJarCmd = LoopCommand.nullary("nextJar", "gets next path of Scio REPL jar",
+    () => { Result.resultFromString(scioClassLoader.get.getNextReplCodeJarPath) } )
+  private val getNewScioContextCmd = LoopCommand.cmd("newScio", "newScio <[scio-context-val-name] | sc>", "gets new Scio context",
+    (scioContextVal) => {
+      val nextReplJar = scioClassLoader.get.getNextReplCodeJarPath
+      intp.beQuietDuring({
+        intp.interpret("val __scio__opts__ = " + args.map('"' + _ + '"'))
+        intp.interpret("val __scio__df__opts__ = PipelineOptionsFactory.fromArgs(__scio__opts__.toArray).as(classOf[DataflowPipelineOptions])")
+
+        scioContextVal match {
+          case v if v.nonEmpty => intp.interpret ("val " + scioContextVal + " = ScioContext(__scio__df__opts__, List(\"" + nextReplJar + "\"))")
+          case _ => intp.interpret ("val sc = ScioContext(__scio__df__opts__, List(\"" + nextReplJar + "\"))")
+        }
+      })
+      Result.default
+    })
+  private val getResultCmd = LoopCommand.cmd("runScio", ":runScio <[scio-context-val-name] | sc>",  "run Scio pipeline",
+    (scioContextVal) => {
+      scioClassLoader.get.createReplCodeJar
+      val scioResult = scioContextVal match {
```

Comment:
scioContextVal is a string right? Match/case is overkill and if/else is easier to read?